spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/ReorderMemory.scala

```diff
@@ -26,50 +26,118 @@ class ReorderMemory(inputFormat: MemoryData, outputFormat: MemoryData,
   _outputFormats = Array(outputFormat)
   _gradInputFormats = Array(gradInputFormat)
 
+  private var realInput : Array[MemoryData] = null
+  private var realOutput : Array[MemoryData] = null
+  private var realgradInput : Array[MemoryData] = null
+  private var realgradOutput : Array[MemoryData] = null
 
   override private[mkldnn] def initFwdPrimitives(inputs: Array[MemoryData], phase: Phase) = {
     _inputFormats = if (inputFormat == null) inputs else Array(inputFormat)
     require(_inputFormats.length == 1, "Only accept one tensor as input")
 
-    require(_inputFormats(0).shape.product == outputFormat.shape.product,
+    if (outputFormat == null) _outputFormats = _inputFormats
+    require(_inputFormats(0).shape.product == _outputFormats(0).shape.product,
       "input output memory not match")
```

Comment:
memory size not matched, also please output the size accordingly

Reply:
ok, changed